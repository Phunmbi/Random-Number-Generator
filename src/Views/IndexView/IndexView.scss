.generator {
  display: flex;
  height: 98vh;
  justify-content: center;
  flex-direction: column;
  &-title{
    margin-top: -100px;
    display: flex;
    justify-content: center;
    font-family: 'Hanalei', cursive;
    font-size: 40px;
    font-weight: 400;
    @media only screen and (max-width: 700px) {
      font-size: 18px;
    }
  }

  form{
    display: flex;
    flex-direction: column;
    width: 340px;
    justify-content: space-around;
    align-content: space-between;
    margin: 0 auto;
    height: 150px;
    @media (max-width: 700px) {
      width: 300px;
    }
    input{
      height: 40px;
      border: 1px solid #4a4a4a;
      border-radius: 5px;
      text-indent: 15px;
      font-family: 'Darker Grotesque', sans-serif;
      font-size: 17px;
      font-weight: bold;
      outline: none;
      @media (max-width: 700px) {
        font-size: 15px;
      }
    }

    .generator-actions {
      display: flex;
      justify-content: space-between;
      input[type="submit"], &__download {
        border: none;
        text-indent: 0;
        background-color: #c6b75d;
        cursor: pointer;
        width: 160px;
        text-align: center;
        color: #000;
        font-size: 17px;
        font-weight: bold;
        @media (max-width: 700px) {
          font-size: 15px;
          width: 140px;
        }
      }

      &__download {
        display: flex;
        justify-content: space-evenly;
        align-items: center;
        width: 160px;
        color: #000;
        font-weight: bold;
        height: 40px;
        border-radius: 5px;
        @media (max-width: 700px) {
          width: 140px;
        }
      }
    }
  }

  hr{
    height: 1px;
    border: none;
    box-shadow: 1px 1px 10px 1px black;
    width: 90vw;
  }
}
