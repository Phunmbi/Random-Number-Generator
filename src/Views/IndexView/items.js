export const items = [
	{
		id: "1111",
		image:
			"https://target.scene7.com/is/image/Target/GUEST_448cd375-9756-4fdc-99e3-96396f30dc29?wid=488&hei=488&fmt=pjpeg",
		quantity: 1,
		title: "Sony Playstation 5",
		price: 35000,
		grams: 3000,
		sku: "Sony Playstation 5 white",
		variantInfo: [
			{
				id: "4444",
				name: "colour",
				value: "white",
			},
		],
	},
	{
		id: "1131",
		image:
			"https://news.xbox.com/en-us/wp-content/uploads/sites/2/2019/12/XboxSeriesXHERO.jpg?fit=1920%2C1080",
		quantity: 1,
		title: "Xbox Series X",
		price: 33000,
		grams: 3000,
		sku: "Xbox Series X black",
		variantInfo: [
			{
				id: "444334",
				name: "colour",
				value: "black",
			},
		],
	},
];
