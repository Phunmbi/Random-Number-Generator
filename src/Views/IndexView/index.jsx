import React, {Component} from 'react';
import './IndexView.scss';

class IndexView extends Component {
  state = {
    amountToGenerate: 1000
  };

  handleChange = (e) => {
    this.setState({
      amountToGenerate: e.target.value
    })
  };

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.history.push(`/generated?${this.state.amountToGenerate}`);
  };

  render() {
    return (
      <div className="generator">
        <div className="generator-title">
          <h2>Random Number Generator</h2>
        </div>
        <form onSubmit={(e) => this.handleSubmit(e)}>
          <input
            type="text"
            value={this.state.amountToGenerate}
            onChange={(e) => this.handleChange(e)}
            className="generator-input"
            placeholder="How many numbers would you like to generate?"/>
          <div className="generator-actions">
            <input type="submit" value="Generate Numbers"/>
          </div>
        </form>
      </div>
    );
  }
}

export default IndexView;
