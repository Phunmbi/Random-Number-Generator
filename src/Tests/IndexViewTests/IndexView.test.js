import React from 'react';
import IndexView from '../../Views/IndexView';

const props = {
  history: {
    push: jest.fn()
  }
};

let wrapped;

beforeEach(() => {
  wrapped = mount (<IndexView {...props} />)
});

it('should take screenshot successfully', () => {
  expect(wrapped).toMatchSnapshot();
});

it('should simulate handle submit', () => {
  wrapped.find('.generator').find('form').simulate('submit');

  const inst = wrapped.instance();

  inst.handleSubmit = jest.fn();

  wrapped.find('.generator').find('form').simulate('submit');
  expect(inst.handleSubmit).toHaveBeenCalled();
});

it('should simulate handle change', () => {
  wrapped.find('.generator').find('form').find('.generator-input') .simulate('change');

  const inst = wrapped.instance();
  inst.handleChange = jest.fn();

  wrapped.find('.generator').find('form').find('.generator-input') .simulate('change');
  expect(inst.handleChange).toHaveBeenCalled();
});
