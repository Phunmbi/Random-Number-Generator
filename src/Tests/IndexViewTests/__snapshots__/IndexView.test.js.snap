// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take screenshot successfully 1`] = `
<IndexView
  history={
    Object {
      "push": [MockFunction],
    }
  }
>
  <div
    className="generator"
  >
    <div
      className="generator-title"
    >
      <h2>
        Random Number Generator
      </h2>
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        className="generator-input"
        onChange={[Function]}
        placeholder="How many numbers would you like to generate?"
        type="text"
        value={0}
      />
      <div
        className="generator-actions"
      >
        <input
          type="submit"
          value="Generate Numbers"
        />
      </div>
    </form>
  </div>
</IndexView>
`;
